src/test/java/teammates/sqllogic/core/CoursesLogicTest.java

```diff
@@ -128,4 +139,300 @@ public void testGetSectionNamesForCourse_courseDoesNotExist_throwEntityDoesNotEx
 
         assertEquals("Trying to get section names for a non-existent course.", ex.getMessage());
     }
+
+    @Test
+    public void testCreateCourse_shouldReturnCreatedCourse_success()
+            throws EntityAlreadyExistsException, InvalidParametersException {
+        Course course = getTypicalCourse();
+
+        when(coursesDb.createCourse(course)).thenReturn(course);
+
+        Course createdCourse = coursesLogic.createCourse(course);
+
+        verify(coursesDb, times(1)).createCourse(course);
+        assertNotNull(createdCourse);
+    }
+
+    @Test
+    public void testCreateDuplicateCourse_throwEntityAlreadyExistsException()
+            throws InvalidParametersException, EntityAlreadyExistsException {
+        Course course = getTypicalCourse();
+
+        when(coursesDb.createCourse(course))
+                .thenThrow(new EntityAlreadyExistsException(
+                    String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS, course.toString())));
+
+        EntityAlreadyExistsException ex = assertThrows(EntityAlreadyExistsException.class,
+                () -> coursesLogic.createCourse(course));
+
+        assertEquals(String.format(ERROR_CREATE_ENTITY_ALREADY_EXISTS, course.toString()), ex.getMessage());
+    }
+
+    @Test
+    public void testGetCourse_shouldReturnCourse_success() {
+        Course course = getTypicalCourse();
+        String courseId = course.getId();
+
+        when(coursesDb.getCourse(courseId)).thenReturn(course);
+
+        Course returnedCourse = coursesLogic.getCourse(courseId);
+
+        verify(coursesDb, times(1)).getCourse(courseId);
+        assertNotNull(returnedCourse);
+    }
+
+    @Test
+    public void testDeleteCourseCascade_shouldDeleteCourse_success() {
+        Course course = getTypicalCourse();
+        List<Instructor> instructors = new ArrayList<>();
+        List<FeedbackSession> feedbackSessions = new ArrayList<>();
+
+        instructors.add(getTypicalInstructor());
+
+        when(fsLogic.getFeedbackSessionsForCourse(course.getId())).thenReturn(feedbackSessions);
+        when(usersLogic.getInstructorsForCourse(course.getId())).thenReturn(instructors);
+
+        when(coursesDb.getCourse(course.getId())).thenReturn(course);
+
+        coursesLogic.deleteCourseCascade(course.getId());
+
+        verify(coursesDb, times(1)).deleteCourse(course);
```

Comment:
we should verify that `fsLogic` and `usersLogic` are called with the appropriate parameters here as well

Reply:
apologies, I should've been more specific here, I meant verifying the parameters for `deleteFeedbackSessionCascade` and `deleteInstructorCascade`, also we should add a verify for `deleteSectionsByCourseId`